server/channels/app/opengraph.go

```diff
@@ -39,6 +41,11 @@ func (a *App) GetOpenGraphMetadata(requestURL string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if len(ogJSON) > openGraphMetadataCacheEntrySizeLimit {
```

Comment:
This is checking the output of `parseOpenGraphMetadata`, but `parseOpenGraphMetadata` is [also called from `getLinkMetadata` via `parseLinkMetadata`](https://github.com/mattermost/mattermost/blob/72b648cf81d94475e57fef123d3ae57ecedbcbb8/server/channels/app/post_metadata.go#L687) and that invocation should apply the same limit, as there's a cache there as well.

Could we move the check inside the `parseOpenGraphMetadata` method instead? That would make sure the check is always applied, and also make testing easier since no `httptest` setup would then be needed to test this.

Reply:
Good idea, done, thanks Juho.

Reply:
@streamer45 I had to change the signature with (I feel) minimal impact. Alternatively, we could not return the error and the marshalled JSON, and log the warning here and then re-marshal later. The problem with that is the double marshalling and that we would be logging warnings for bad links (which we have just been ignoring). I think I prefer changing the signature, but I'll defer to you on this.